tests/e2e/vmgroupbasic_test.go

```diff
@@ -0,0 +1,240 @@
+	// as access-rights have been changed to True
+	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeNames[1]+"@"+s.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Check if volume was successfully created
+	isAvailable = verification.CheckVolumeAvailability(s.config.DockerHosts[1], s.volumeNames[1]+"@"+s.config.Datastores[1])
+	c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", s.volumeNames[1]))
+
+	// verify able to delete  volume
+	out, err = dockercli.DeleteVolume(s.config.DockerHosts[1], s.volumeNames[1]+"@"+s.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Remove the create privilege on the non-default vmgroup for specified datastore
+	admincli.RemoveCreateAccessForVMgroup(s.config.EsxHost, vmGroupName, s.config.Datastores[1])
+
+	// Create a volume from non-default vmgroup's vm - operation should fail
+	volumeName := inputparams.GetUniqueVolumeName(vmGroupName)
+	out, err = dockercli.CreateVolume(s.config.DockerHostNames[1], volumeName+"@"+s.config.Datastores[1])
+	c.Assert(err, Not(IsNil), Commentf(out))
+
+	out, err = dockercli.DeleteVolume(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	admincli.RemoveVMFromVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[1])
+	misc.LogTestEnd(c.TestName())
+}
+
+// Test step:
+// 1. Create a vmgroup and associate a vm to it.
+// 2. Execute vmgroup ls command.
+// 3. Kill the vm that was added to the tenant.
+// 4. Again execute vmgroup ls command to verify command works fine.
+func (s *vgBasicSuite) TestVmgroupLsPostVMKill(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	admincli.CreateVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[0], admincliconst.VMHomeDatastore)
+	isVmgroupAvailable := admincli.CheckVmgroupAvailability(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup ls command does not lists the vmgroup %s .", vmGroupName))
+
+	powerState := esx.GetVMPowerState(s.config.DockerHostNames[0])
+	log.Printf("VM[%s]'s current power state is [%s]", s.config.DockerHostNames[0], properties.PowerOnState)
+	c.Assert(powerState, Equals, properties.PowerOnState, Commentf("VM [%s] should be powered on state", s.config.DockerHostNames[0]))
+
+	vmProcessID := esx.GetVMProcessID(s.config.EsxHost, s.config.DockerHostNames[0])
+	log.Printf("VM's process ID is: %s", vmProcessID)
+
+	isVMKilled := esx.KillVM(s.config.EsxHost, vmProcessID)
+	c.Assert(isVMKilled, Equals, true, Commentf("Unable to kill VM %s ...", s.config.DockerHostNames[0]))
+
+	isStatusChanged := esx.WaitForExpectedState(esx.GetVMPowerState, s.config.DockerHostNames[0], properties.PowerOffState)
+	c.Assert(isStatusChanged, Equals, true, Commentf("VM [%s] should be powered off state", s.config.DockerHostNames[0]))
+
+	isVmgroupAvailable = admincli.CheckVmgroupAvailability(s.config.EsxHost, vmGroupName)
```

Comment:
@govint Basically this test tries to automate this scenario: PR #929 
After the vm was killed, ```vmgroup ls``` used to throw stacktrace.